ui/pages/asset/asset.tsx

```diff
@@ -3,26 +3,43 @@ import { useSelector } from 'react-redux';
 import { Redirect, useParams } from 'react-router-dom';
 import { isEqualCaseInsensitive } from '../../../shared/modules/string-utils';
 import NftDetails from '../../components/app/assets/nfts/nft-details/nft-details';
-import {
-  getNativeCurrency,
-  getNfts,
-  getTokens,
-} from '../../ducks/metamask/metamask';
+import { getSelectedAccountTokensAcrossChains } from '../../selectors';
+import { getNativeCurrency, getNfts } from '../../ducks/metamask/metamask';
 import { DEFAULT_ROUTE } from '../../helpers/constants/routes';
 
 import NativeAsset from './components/native-asset';
 import TokenAsset from './components/token-asset';
 
+function findAssetByAddress(data: any, address?: string) {
```

Comment:
We should move this to a helper file so we can add tests

Reply:
Moved to util file here: https://github.com/MetaMask/metamask-extension/pull/28193/commits/ebac7f2537fbed36464ce5c23a516676658838c0

Added a TODO for the unit tests. We'll need to do a couple passovers on the feature branch for this.